Sources/NIO/BaseSocket.swift

```diff
@@ -12,6 +12,28 @@
 //
 //===----------------------------------------------------------------------===//
 
+import NIOConcurrencyHelpers
+
+private let globallyIgnoredSIGPIPE: Atomic<Bool> = .init(value: false)
+
+func ignoreSIGPIPE(descriptor fd: CInt) throws {
+    if !globallyIgnoredSIGPIPE.load() {
+        #if os(Linux)
+        /* no F_SETNOSIGPIPE on Linux :( */
+        _ = unsafeBitCast(Glibc.signal(SIGPIPE, SIG_IGN) as sighandler_t?, to: Int.self)
+        globallyIgnoredSIGPIPE.store(true)
+        #else
+        assert(fd >= 0, "illegal file descriptor \(fd)")
```

Comment:
May as well be a `precondition` tbh.

Reply:
@Lukasa usually we do 'if it's a bug in NIO that doesn't lead to disaster' == `assert`, no? This doesn't cause any harm if `< 0`, `fcntl` will return `EBADF`.

Reply:
@Lukasa wdyt?

Reply:
Happy to ignore this then.